code/part-two/README.md

```diff
@@ -1,54 +1,589 @@
+spaces._
+
+#### Collection Prefix
+
+For both cryptomoji and offers (see extra credit), the next 8 characters are a
+prefix for the collection they belong to. These are generated from the first
+eight characters of a SHA-512 hash of the collection’s public key.
+
+```
+1b96dbb5
+```
+
+#### Cryptomoji
+
+The final 54 characters of a cryptomoji’s address is the first 54 characters of
+a SHA-512 hash of their DNA string.
+
+```
+5f4d76 01 1b96dbb5 d1dd9beeb54338f0650588247d5b14af9609a59a9df42a61cfa7b7
+```
+
+#### Sire Listing
+
+The last 62 characters of a Sire Listing’s address are the first 62 characters
+of a SHA-512 hash of the owner’s public key.
+
+```
+5f4d76 02 1b96dbb5322e410816dd41d93571801e751a4f0cc455d8bd58f5f8ad3d67cb
+```
+
+### Payloads
+
+Cryptomoji payloads will be objects, each with an `"action"` key that
+designates what event should occur in the transaction processor. Each actions
+will be spelled out by a specific string written in CONSTANT_CASE. Any
+additional data required, will be included in other keys particular to that
+payload.
+
+#### Create Collection
+
+```json
+{
+    "action": "CREATE_COLLECTION"
+}
+```
+
+_Validation:_
+- Signer must not already have a collection
+
+Creates a new collection for the signer of the transaction with three new
+pseudo-random (but deterministic!) cryptomoji. Like other actions, the identity
+of the signer will come from their public key in the transaction header, and so
+is not included in the payload itself.
+
+#### Select Sire
+
+```json
+{
+    "action": "SELECT_SIRE",
+    "sire": "<string, moji address>"
+}
+```
+
+_Validation:_
+- Signer must have a collection
+- The cryptomoji must exist
+- Signer must own the sire
+
+Lists a cryptomoji as a collection's sire, indicating it is available for other
+collections to breed with. Collections may only have one sire at a time.
+
+#### Breed Moji
+
+```json
+{
+    "action": "BREED_MOJI",
+    "sire": "<string, moji address>",
+    "breeder": "<string, moji address>"
+}
+```
+
+_Validation:_
+- Signer must have a collection
+- The cryptomoji must exist
+- Signer must own the breeder
+- Sire must be listed as a sire
+
+Creates a new cryptomoji for the owner of the _breeder_, which is a
+pseudo-random combination of the DNA from the breeder and the sire.
+
+## Extra Credit
+
+For extra credit, add the capability for collections to trade cryptomoji
+between each other. This is harder than it sounds. One user will need to create
+an offer, while others add responses until the offer owner accepts them. This
+is a process that spans multiple transactions, never mind the possibility that
+one party might decide to cancel their offer or response.
+
+### State Entities
+
+#### Offer
+
+```json
+{
+    "owner": <string, public key>,
+    "moji": [ <strings, moji addresses> ],
+    "responses": [
+        {
+            "approver": <string, public key>,
+            "moji": [ <strings, moji addresses> ]
```

Comment:
I think lns 450, 451, 454, 455 should have double quotes around the values if we want to be consistent w/ how we represented this json information previously
```json
{
    "owner": "<string, public key>",
    "moji": [ "<strings, moji addresses>" ],
    "responses": [
        {
            "approver": "<string, public key>",
            "moji": [ "<strings, moji addresses>" ]
        }
    ]
}
```